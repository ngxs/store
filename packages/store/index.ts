/**
 * The public api for consumers of @ngxs/store
 */
export * from './src/public_api';

/**
 * The plugin api for the stuff that a plugins needs
 */
export * from './src/plugin_api';
