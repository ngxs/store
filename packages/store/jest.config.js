module.exports = {
  displayName: 'store',
  preset: '../../jest.preset.js',
  setupFilesAfterEnv: ['<rootDir>/test-setup.ts'],
  globals: {},
  coverageDirectory: '../../coverage/packages/store',
  coveragePathIgnorePatterns: ['/node_modules/', '<rootDir>/schematics/src/utils/ng-utils/'],
  transform: {
    '^.+\\.(ts|mjs|js|html)$': [
      'jest-preset-angular',
      {
        isolatedModules: true,
        tsconfig: '<rootDir>/tsconfig.spec.json',
        stringifyContentPathRegex: '\\.(html|svg)$'
      }
    ]
  },
  snapshotSerializers: [
    'jest-preset-angular/build/serializers/no-ng-attributes',
    'jest-preset-angular/build/serializers/ng-snapshot',
    'jest-preset-angular/build/serializers/html-comment'
  ]
};
