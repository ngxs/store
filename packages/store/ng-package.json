{
  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../builds/store",
  "lib": {
    "flatModuleFile": "ngxs-store",
    "entryFile": "index.ts",
    "umdModuleIds": {}
  }
}
