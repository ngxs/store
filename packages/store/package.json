{
  "$schema": "../../node_modules/ng-packagr/package.schema.json",
  "name": "@ngxs/store",
  "version": "0.0.0",
  "license": "MIT",
  "sideEffects": true,
  "peerDependencies": {
    "@angular/core": ">=5.0.0 <8.0.0",
    "rxjs": ">=6.0.0 || ^5.6.0-forward-compat.4"
  },
  "ngPackage": {
    "lib": {
      "flatModuleFile": "ngxs-store",
      "entryFile": "index.ts",
      "umdModuleIds": {
        "@ngxs/store/internals": "ngxs-store-internals",
        "@ngxs/store/operators": "ngxs-store-operators"
      }
    },
    "dest": "../../@ngxs/store"
  }
}
