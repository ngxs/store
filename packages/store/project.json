{
  "name": "store",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "root": "packages/store",
  "sourceRoot": "packages/store",
  "projectType": "library",
  "targets": {
    "build": {
      "executor": "@angular-devkit/build-angular:ng-packagr",
      "outputs": ["@ngxs/store"],
      "options": {
        "tsConfig": "tsconfig.build.json",
        "project": "packages/store/ng-package.json"
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "packages/store/jest.config.js",
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true
        }
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": ["packages/store/**/*.ts", "packages/store/**/*.html"]
      }
    }
  }
}
