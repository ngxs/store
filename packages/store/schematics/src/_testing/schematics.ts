import { SchematicTestRunner, UnitTestTree } from '@angular-devkit/schematics/testing';
import { workspaceRoot } from '@nrwl/devkit';
import { Schema as ApplicationOptions } from '@schematics/angular/application/schema';
import { Schema as WorkspaceOptions } from '@schematics/angular/workspace/schema';
import * as path from 'path';

const angularSchematicRunner = new SchematicTestRunner(
  '@schematics/angular',
  path.join(workspaceRoot, 'node_modules/@schematics/angular/collection.json')
);

const defaultWorkspaceOptions: WorkspaceOptions = {
  name: 'workspace',
  newProjectRoot: 'projects',
  version: '1.0.0'
};

const defaultAppOptions: ApplicationOptions = {
  name: 'foo',
  inlineStyle: false,
  inlineTemplate: false,
  routing: true,
  skipTests: false,
  skipPackageJson: false
};

export async function createWorkspace(
  isStandalone = false,
  schematicRunner = angularSchematicRunner,
  workspaceOptions = defaultWorkspaceOptions,
  appOptions = defaultAppOptions
) {
  let appTree: UnitTestTree = await schematicRunner.runSchematic(
    'workspace',
    workspaceOptions
  );
  appTree = await schematicRunner.runSchematic('application', appOptions, appTree);

  isStandalone && updateToStandalone(appTree, workspaceOptions, appOptions);

  return appTree;
}

// Note: This is a workaround to convert the application as standalone. Should be removed when migrating to NG17
function updateToStandalone(
  appTree: UnitTestTree,
  workspaceOptions: WorkspaceOptions,
  appOptions: ApplicationOptions
) {
  const mainTsContent = `
      import {
        bootstrapApplication,
      } from '@angular/platform-browser';
      import { AppComponent } from './app/app.component';

      bootstrapApplication(AppComponent).catch((err) =>
        console.error(err),
      );
  `;

  const appComponentContent = `
      import { Component } from '@angular/core';

      @Component({
        selector: 'app-root',
        standalone: true,
        templateUrl: './app.component.html',
        styleUrls: ['./app.component.scss'],
      })
      export class AppComponent {}
  `;

  const projectPath = `/${workspaceOptions.newProjectRoot}/${appOptions.name}`;
  appTree.overwrite(`${projectPath}/src/main.ts`, mainTsContent);
  appTree.overwrite(`${projectPath}/src/app/app.component.ts`, appComponentContent);
  appTree.delete(`${projectPath}/src/app/app.module.ts`);
}
