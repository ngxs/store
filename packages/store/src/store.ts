import { Injectable, NgZone } from '@angular/core';
import { Observable, of, Subscription } from 'rxjs';
import { catchError, distinctUntilChanged, map, take } from 'rxjs/operators';

import { getSelectorFn } from './utils/selector-utils';
import { InternalStateOperations } from './internal/state-operations';
import { StateStream } from './internal/state-stream';
import { enterZone } from './operators/zone';

@Injectable()
export class Store {
  constructor(
    private _ngZone: NgZone,
    private _stateStream: StateStream,
    private _internalStateOperations: InternalStateOperations
  ) {}

  /**
   * Dispatches event(s).
   */
  dispatch(event: any | any[]): Observable<any> {
    return this._internalStateOperations.getRootStateOperations().dispatch(event);
  }

  /**
   * Selects a slice of data from the store.
   */
  select<T>(selector: (state: any, ...states: any[]) => T): Observable<T>;
  select(selector: string | any): Observable<any>;
  select(selector: any): Observable<any> {
    const selectorFn = getSelectorFn(selector);
    return this._stateStream.pipe(
      map(selectorFn),
      catchError(err => {
        // if error is TypeError we swallow it to prevent usual errors with property access
        if (err instanceof TypeError) {
          return of(undefined);
        }

        // rethrow other errors
        throw err;
      }),
      distinctUntilChanged(),
      enterZone(this._ngZone)
    );
  }

  /**
   * Select one slice of data from the store.
   */
  selectOnce<T>(selector: (state: any, ...states: any[]) => T): Observable<T>;
  selectOnce(selector: string | any): Observable<any>;
  selectOnce(selector: any): Observable<any> {
    return this.select(selector).pipe(take(1));
  }

  /**
   * Select a snapshot from the state.
   */
  selectSnapshot<T>(selector: (state: any, ...states: any[]) => T): T;
  selectSnapshot(selector: string | any): any;
  selectSnapshot(selector: any): any {
    const selectorFn = getSelectorFn(selector);
    return selectorFn(this._stateStream.getValue());
  }

  /**
   * Allow the user to subscribe to the root of the state
   */
  subscribe(fn?: any): Subscription {
    return this._stateStream.pipe(enterZone(this._ngZone)).subscribe(fn);
  }

  /**
   * Return the raw value of the state.
   */
  snapshot(): any {
    return this._internalStateOperations.getRootStateOperations().getState();
  }

  /**
   * Reset the state to a specific point in time. This method is useful
   * for plugin's who need to modify the state directly or unit testing.
   */
  reset(state: any) {
    return this._internalStateOperations.getRootStateOperations().setState(state);
  }
}
