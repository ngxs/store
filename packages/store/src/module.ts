import { NgModule, ModuleWithProviders, Optional, Inject } from '@angular/core';

import { ROOT_STATE_TOKEN, FEATURE_STATE_TOKEN } from './symbols';
import { StateFactory } from './state-factory';
import { Actions } from './actions-stream';
import { Store } from './store';
import { SelectFactory } from './select';
import { StateStream } from './state-stream';
import { PluginManager } from './plugin-manager';
import { InitState, UpdateState } from './actions';

@NgModule()
export class NgxsRootModule {
  constructor(
    private _factory: StateFactory,
    private _stateStream: StateStream,
    store: Store,
    select: SelectFactory,
    @Optional()
    @Inject(ROOT_STATE_TOKEN)
    states: any[]
  ) {
    this.initStates(states);
    store.dispatch(new InitState());
  }

  initStates(states) {
    if (states) {
      // add stores to the state graph and return their defaults
      const init = this._factory.addAndReturnDefaults(states);

      // get our current stream
      const cur = this._stateStream.getValue();

      // set the state to the current + new
      this._stateStream.next({ ...cur, ...init });
    }
  }
}

@NgModule({})
export class NgxsFeatureModule {
  constructor(
    root: NgxsRootModule,
    store: Store,
    private _stateStream: StateStream,
    private _factory: StateFactory,
    @Optional()
    @Inject(FEATURE_STATE_TOKEN)
    states: any[][]
  ) {
    // since FEATURE_STATE_TOKEN is a multi token, we need to
    // flatten it [[Feature1State, Feature2State], [Feature3State]]
    const flattenedStates = ([] as any[]).concat(...states);
    this.initStates(flattenedStates);
    store.dispatch(new UpdateState());
  }

  initStates(stores) {
    if (stores) {
      // bind the stores
      const init = this._factory.addAndReturnDefaults(stores);

      // get our current stream
      const cur = this._stateStream.getValue();

      // set the state to the current + new
      this._stateStream.next({ ...cur, ...init });
    }
  }
}

@NgModule({})
export class NgxsModule {
  static forRoot(states: any[] = []): ModuleWithProviders {
    return {
      ngModule: NgxsRootModule,
      providers: [
        StateFactory,
        Actions,
        Store,
        StateStream,
        SelectFactory,
        PluginManager,
        ...states,
        {
          provide: ROOT_STATE_TOKEN,
          useValue: states
        }
      ]
    };
  }

  static forFeature(states: any[]): ModuleWithProviders {
    return {
      ngModule: NgxsFeatureModule,
      providers: [
        StateFactory,
        PluginManager,
        ...states,
        {
          provide: FEATURE_STATE_TOKEN,
          multi: true,
          useValue: states
        }
      ]
    };
  }
}
