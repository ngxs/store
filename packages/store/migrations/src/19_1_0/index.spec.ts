import { SchematicTestRunner } from '@angular-devkit/schematics/testing';
import { workspaceRoot } from '@nrwl/devkit';
import { join } from 'path';
import { createWorkspace } from '../../../schematics-utils/_testing';

const versions = require('../../../schematics-utils/src/versions.json');
const ngxsStoreVersion: string = versions['@ngxs/store'];

describe('Migrate to 19.1.0', () => {
  const ngxsMigrationsRunner = new SchematicTestRunner(
    '@ngxs/store/migrations',
    join(workspaceRoot, 'packages/store/migrations/migrations.json')
  );

  const testSetup = async (options?: { isStandalone?: boolean }) => {
    const appTree = await createWorkspace(options?.isStandalone);
    appTree.overwrite(
      'package.json',
      `{"dependencies": {"@ngxs/store": "${ngxsStoreVersion}"}}`
    );

    return { appTree };
  };

  describe('NgxsModule.forRoot', () => {
    it('migrate empty forRoot', async () => {
      const { appTree } = await testSetup();

      const newContent = `
      import { NgxsModule } from '@ngxs/store';
      import { NgModule } from '@angular/core';
      import { BrowserModule } from '@angular/platform-browser';

      import { AppRoutingModule } from './app-routing.module';
      import { AppComponent } from './app.component';

        @NgModule({
          declarations: [
            AppComponent
          ],
          imports: [
            BrowserModule,
            AppRoutingModule,
            NgxsModule.forRoot()
          ],
          bootstrap: [AppComponent]
        })
        export class AppModule { }
      `;

      appTree.overwrite('/projects/foo/src/app/app.module.ts', newContent);

      const tree = await ngxsMigrationsRunner.runSchematic(
        'ngxs-store-migration-19-1-0',
        {},
        appTree
      );

      const contentUpdate = tree!.readContent('/projects/foo/src/app/app.module.ts');
      expect(contentUpdate).toMatchSnapshot();
    });

    it('migrate forRoot with states', async () => {
      const { appTree } = await testSetup();

      const newContent = `
      import { NgxsModule } from '@ngxs/store';
      import { NgModule } from '@angular/core';
      import { BrowserModule } from '@angular/platform-browser';

      import { AppRoutingModule } from './app-routing.module';
      import { AppComponent } from './app.component';

        @NgModule({
          declarations: [
            AppComponent
          ],
          imports: [
            BrowserModule,
            AppRoutingModule,
            NgxsModule.forRoot(states)
          ],
          bootstrap: [AppComponent]
        })
        export class AppModule { }
      `;

      appTree.overwrite('/projects/foo/src/app/app.module.ts', newContent);

      const tree = await ngxsMigrationsRunner.runSchematic(
        'ngxs-store-migration-19-1-0',
        {},
        appTree
      );

      const contentUpdate = tree!.readContent('/projects/foo/src/app/app.module.ts');
      expect(contentUpdate).toMatchSnapshot();
    });

    it('migrate forRoot with states and existing options', async () => {
      const { appTree } = await testSetup();

      const newContent = `
      import { NgxsModule } from '@ngxs/store';
      import { NgModule } from '@angular/core';
      import { BrowserModule } from '@angular/platform-browser';

      import { AppRoutingModule } from './app-routing.module';
      import { AppComponent } from './app.component';

        @NgModule({
          declarations: [
            AppComponent
          ],
          imports: [
            BrowserModule,
            AppRoutingModule,
            NgxsModule.forRoot(states, {foo:'bar'})
          ],
          bootstrap: [AppComponent]
        })
        export class AppModule { }
      `;

      appTree.overwrite('/projects/foo/src/app/app.module.ts', newContent);

      const tree = await ngxsMigrationsRunner.runSchematic(
        'ngxs-store-migration-19-1-0',
        {},
        appTree
      );

      const contentUpdate = tree!.readContent('/projects/foo/src/app/app.module.ts');
      expect(contentUpdate).toMatchSnapshot();
    });

    it('migrate forRoot from a constant array', async () => {
      const { appTree } = await testSetup();

      const newContent = `
      import { NgxsModule } from '@ngxs/store';
      import { NgModule } from '@angular/core';
      import { BrowserModule } from '@angular/platform-browser';

      import { AppRoutingModule } from './app-routing.module';
      import { AppComponent } from './app.component';

      const imports = [
        BrowserModule,
        AppRoutingModule,
        NgxsModule.forRoot(states, {foo:'bar'})
      ];

        @NgModule({
          declarations: [
            AppComponent
          ],
          imports,
          bootstrap: [AppComponent]
        })
        export class AppModule { }
      `;

      appTree.overwrite('/projects/foo/src/app/app.module.ts', newContent);

      const tree = await ngxsMigrationsRunner.runSchematic(
        'ngxs-store-migration-19-1-0',
        {},
        appTree
      );

      const contentUpdate = tree!.readContent('/projects/foo/src/app/app.module.ts');
      expect(contentUpdate).toMatchSnapshot();
    });
  });

  describe('provideStore', () => {
    it('migrate provideStore with states without options', async () => {
      const { appTree } = await testSetup({
        isStandalone: true
      });

      const newContent = `
      import { bootstrapApplication } from '@angular/platform-browser';
      import { AppComponent } from './app/app.component';
      import { provideStore } from '@ngxs/store';

      bootstrapApplication(AppComponent, {
        providers: [provideStore([])],
      });

      `;

      appTree.overwrite('/projects/foo/src/main.ts', newContent);

      const tree = await ngxsMigrationsRunner.runSchematic(
        'ngxs-store-migration-19-1-0',
        {},
        appTree
      );

      const contentUpdate = tree!.readContent('/projects/foo/src/main.ts');
      expect(contentUpdate).toMatchSnapshot();
    });

    it('migrate provideStore with states and existing options', async () => {
      const { appTree } = await testSetup({
        isStandalone: true
      });

      const newContent = `
      import { bootstrapApplication } from '@angular/platform-browser';
      import { AppComponent } from './app/app.component';
      import { provideStore } from '@ngxs/store';

      bootstrapApplication(AppComponent, {
        providers: [provideStore([], {foo:'bar'})],
      });

      `;

      appTree.overwrite('/projects/foo/src/main.ts', newContent);

      const tree = await ngxsMigrationsRunner.runSchematic(
        'ngxs-store-migration-19-1-0',
        {},
        appTree
      );

      const contentUpdate = tree!.readContent('/projects/foo/src/main.ts');

      expect(contentUpdate).toMatchSnapshot();
    });

    it('migrate provideStore with states, options and plugins', async () => {
      const { appTree } = await testSetup({
        isStandalone: true
      });

      const newContent = `
      import { bootstrapApplication } from '@angular/platform-browser';
      import { AppComponent } from './app/app.component';
      import { provideStore } from '@ngxs/store';

      bootstrapApplication(AppComponent, {
        providers: [provideStore([], {foo:'bar'}, withNgxsRouterPlugin(), withNgxsStoragePlugin())],
      });

      `;

      appTree.overwrite('/projects/foo/src/main.ts', newContent);

      const tree = await ngxsMigrationsRunner.runSchematic(
        'ngxs-store-migration-19-1-0',
        {},
        appTree
      );

      const contentUpdate = tree!.readContent('/projects/foo/src/main.ts');

      expect(contentUpdate).toMatchSnapshot();
    });

    it('migrate provideStore with states and plugins', async () => {
      const { appTree } = await testSetup({
        isStandalone: true
      });

      const newContent = `
      import { bootstrapApplication } from '@angular/platform-browser';
      import { AppComponent } from './app/app.component';
      import { provideStore } from '@ngxs/store';

      bootstrapApplication(AppComponent, {
        providers: [provideStore([], withNgxsRouterPlugin(), withNgxsStoragePlugin())],
      });

      `;

      appTree.overwrite('/projects/foo/src/main.ts', newContent);

      const tree = await ngxsMigrationsRunner.runSchematic(
        'ngxs-store-migration-19-1-0',
        {},
        appTree
      );

      const contentUpdate = tree!.readContent('/projects/foo/src/main.ts');

      expect(contentUpdate).toMatchSnapshot();
    });
  });
});
