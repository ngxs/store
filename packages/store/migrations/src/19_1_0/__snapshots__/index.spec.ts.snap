// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Migrate to 19.1.0 NgxsModule.forRoot migrate empty forRoot 1`] = `
"
      import { NgxsModule, DispatchOutsideZoneNgxsExecutionStrategy } from '@ngxs/store';
      import { NgModule } from '@angular/core';
      import { BrowserModule } from '@angular/platform-browser';

      import { AppRoutingModule } from './app-routing.module';
      import { AppComponent } from './app.component';

        @NgModule({
          declarations: [
            AppComponent
          ],
          imports: [
            BrowserModule,
            AppRoutingModule,
            NgxsModule.forRoot([], { executionStrategy: DispatchOutsideZoneNgxsExecutionStrategy })
          ],
          bootstrap: [AppComponent]
        })
        export class AppModule { }
      "
`;

exports[`Migrate to 19.1.0 NgxsModule.forRoot migrate forRoot from a constant array 1`] = `
"
      import { NgxsModule, DispatchOutsideZoneNgxsExecutionStrategy } from '@ngxs/store';
      import { NgModule } from '@angular/core';
      import { BrowserModule } from '@angular/platform-browser';

      import { AppRoutingModule } from './app-routing.module';
      import { AppComponent } from './app.component';

      const imports = [
        BrowserModule,
        AppRoutingModule,
        NgxsModule.forRoot(states, {foo:'bar', executionStrategy: DispatchOutsideZoneNgxsExecutionStrategy})
      ];

        @NgModule({
          declarations: [
            AppComponent
          ],
          imports,
          bootstrap: [AppComponent]
        })
        export class AppModule { }
      "
`;

exports[`Migrate to 19.1.0 NgxsModule.forRoot migrate forRoot with states 1`] = `
"
      import { NgxsModule, DispatchOutsideZoneNgxsExecutionStrategy } from '@ngxs/store';
      import { NgModule } from '@angular/core';
      import { BrowserModule } from '@angular/platform-browser';

      import { AppRoutingModule } from './app-routing.module';
      import { AppComponent } from './app.component';

        @NgModule({
          declarations: [
            AppComponent
          ],
          imports: [
            BrowserModule,
            AppRoutingModule,
            NgxsModule.forRoot(states, { executionStrategy: DispatchOutsideZoneNgxsExecutionStrategy })
          ],
          bootstrap: [AppComponent]
        })
        export class AppModule { }
      "
`;

exports[`Migrate to 19.1.0 NgxsModule.forRoot migrate forRoot with states and existing options 1`] = `
"
      import { NgxsModule, DispatchOutsideZoneNgxsExecutionStrategy } from '@ngxs/store';
      import { NgModule } from '@angular/core';
      import { BrowserModule } from '@angular/platform-browser';

      import { AppRoutingModule } from './app-routing.module';
      import { AppComponent } from './app.component';

        @NgModule({
          declarations: [
            AppComponent
          ],
          imports: [
            BrowserModule,
            AppRoutingModule,
            NgxsModule.forRoot(states, {foo:'bar', executionStrategy: DispatchOutsideZoneNgxsExecutionStrategy})
          ],
          bootstrap: [AppComponent]
        })
        export class AppModule { }
      "
`;

exports[`Migrate to 19.1.0 provideStore migrate provideStore with states and existing options 1`] = `
"
      import { bootstrapApplication } from '@angular/platform-browser';
      import { AppComponent } from './app/app.component';
      import { provideStore, DispatchOutsideZoneNgxsExecutionStrategy } from '@ngxs/store';

      bootstrapApplication(AppComponent, {
        providers: [provideStore([], {foo:'bar', executionStrategy: DispatchOutsideZoneNgxsExecutionStrategy})],
      });

      "
`;

exports[`Migrate to 19.1.0 provideStore migrate provideStore with states and plugins 1`] = `
"
      import { bootstrapApplication } from '@angular/platform-browser';
      import { AppComponent } from './app/app.component';
      import { provideStore, DispatchOutsideZoneNgxsExecutionStrategy } from '@ngxs/store';

      bootstrapApplication(AppComponent, {
        providers: [provideStore([], { executionStrategy: DispatchOutsideZoneNgxsExecutionStrategy }, withNgxsRouterPlugin(), withNgxsStoragePlugin())],
      });

      "
`;

exports[`Migrate to 19.1.0 provideStore migrate provideStore with states without options 1`] = `
"
      import { bootstrapApplication } from '@angular/platform-browser';
      import { AppComponent } from './app/app.component';
      import { provideStore, DispatchOutsideZoneNgxsExecutionStrategy } from '@ngxs/store';

      bootstrapApplication(AppComponent, {
        providers: [provideStore([], { executionStrategy: DispatchOutsideZoneNgxsExecutionStrategy })],
      });

      "
`;

exports[`Migrate to 19.1.0 provideStore migrate provideStore with states, options and plugins 1`] = `
"
      import { bootstrapApplication } from '@angular/platform-browser';
      import { AppComponent } from './app/app.component';
      import { provideStore, DispatchOutsideZoneNgxsExecutionStrategy } from '@ngxs/store';

      bootstrapApplication(AppComponent, {
        providers: [provideStore([], {foo:'bar', executionStrategy: DispatchOutsideZoneNgxsExecutionStrategy}, withNgxsRouterPlugin(), withNgxsStoragePlugin())],
      });

      "
`;
